### PR TITLE
Setup Associations

### PR DESCRIPTION
**Pull request by yash244466666**
### In this PR :

- Create a class `Classroom` with the `@label` instance variable.

- Create the `has-many/belongs-to relationship` between `Classroom `and `Student`.

- Create a class `Book `with `@title` and `@author` instance variables.

- Create a class `Rental `with `@date`  instance variable.

- Create the `many-to-many/has-many-through` relationship between `Person `and `Book `using the intermediate class `Rental`.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person, date)
+    rental = Rental.new(date, self, person)
+    @rentals << rental
+    rental
+  end
```
Comment (julie-ify) [inline book.rb]: - [ ] There is really no need for the last two lines in this method. Instantiating a rental object already adds the rental to the respective `rentals` arrays so there is no need to repeat ourselves here.

- [ ] We should also take note of the arrangement of the parameters - `person` should come before `book` as defined in the Rental class. 

  ```
  def add_rental(person, date)
    Rental.new(date, person, book)
  end
  ```

File: person.rb [hunk 2]
```diff
@@ -31,6 +32,10 @@ def generate_unique_id
   def correct_name
     @name
   end
+
+  def add_rental(book, date)
+    Rental.new(date, book, self)
+  end
```
Comment (julie-ify) [inline person.rb]: - [ ] Let us re-arrange the arguments here to match the Rental constructor.

  ```
  def add_rental(book, date)
    Rental.new(date, self, book)
  end
  ```

File: student.rb [hunk 3]
```diff
@@ -6,6 +6,7 @@ class Student < Person
   def initialize(age, classroom, name, parent_permission: true)
     super(age, name, parent_permission)
     @classroom = classroom
+    @classroom.add_student(self)
```
Comment (julie-ify) [inline student.rb]: - [ ] This is not what we want to establish the relationship between student and classroom. The association should be added through its own method. Kindly remove this line of code.

  ```
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @yash244466666,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by yash244466666** [comment]
### Added changes as instructed

- In [book.rb](https://github.com/yash244466666/OOP_School_Library/pull/4/files/d4826caff6c16e234d431e2a282074ae93e2bcf3#diff-7ba27ac0ffd7d61b0e014d8610caf4f6f08c85d452871e71e6959d762b4c960f) removed lines as instructed and re-arranged the parameters
- In [person.rb](https://github.com/yash244466666/OOP_School_Library/pull/4/files/d4826caff6c16e234d431e2a282074ae93e2bcf3#diff-47720676886debe217f2360ee91e7dcbfca82989297cf253a11792456731033f) re-arranged the arguments
- In [student.rb](https://github.com/yash244466666/OOP_School_Library/pull/4/files/d4826caff6c16e234d431e2a282074ae93e2bcf3#diff-520da6d5aea195e484c52d74ddf60964055847deb2bcbf89dfd2a8479e5ce63e) removed the line of code as instructed.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @yash244466666, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Clean and Organized code
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**